### PR TITLE
Add fix/lint helper dev targets

### PR DESCRIPTION
**Pull request by nschonni**
ESLint and Stylelint both allow a `--fix` parameter to autofix rules, but things like Prettier use another `--write` command.
This adds consistent targets so developers can use the tool's autofixing easily, along with a rollup `yarn fix` to run all autofixes.

### CODE REVIEW

**Review by ClearlyClaire — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -5,18 +5,23 @@
...
     "start": "node ./streaming/index.js",
-    "test": "${npm_execpath} run test:lint:js && ${npm_execpath} run test:jest",
-    "test:lint": "${npm_execpath} run test:lint:js && ${npm_execpath} run test:lint:sass",
-    "test:lint:js": "eslint --ext=.js,.jsx . --cache --report-unused-disable-directives",
-    "test:lint:sass": "stylelint \"**/*.{css,scss}\" && prettier --check \"**/*.{css,scss}\"",
+    "test": "npm run test:lint && npm run test:jest",
     "test:jest": "cross-env NODE_ENV=test jest",
-    "format": "prettier --write .",
-    "format-check": "prettier --check ."
+    "test:lint": "npm run test:lint:js && npm run test:lint:json && npm run test:lint:sass && npm run test:lint:yml",
```
Comment (nschonni) [inline package.json:15]: Should i use `yarn` instead of `npm` for these calls?

**Comment by nschonni** [comment]
I updated this to include Markdown targets after #21972 landed, but realized I missed a trigger there, so added in a slightly unrelated commit to fix that

**Review by ClearlyClaire — APPROVED** [review]

**Comment by nschonni** [comment]
If there is a preference to go with the shorter names like in https://github.com/mastodon/mastodon/pull/23920 I can flatter them down to `lint:*` / `fix:*` instead.

File: package.json [hunk 2]
```diff
@@ -5,18 +5,25 @@
     "node": ">=14"
   },
   "scripts": {
-    "postversion": "git push --tags",
     "build:development": "cross-env RAILS_ENV=development NODE_ENV=development ./bin/webpack",
     "build:production": "cross-env RAILS_ENV=production NODE_ENV=production ./bin/webpack",
     "manage:translations": "node ./config/webpack/translationRunner.js",
+    "postversion": "git push --tags",
     "start": "node ./streaming/index.js",
-    "test": "${npm_execpath} run test:lint:js && ${npm_execpath} run test:jest",
```
Comment (ykzts) [inline package.json:13]: Is there a reason you are replacing `${npm_execpath}` with `yarn`?

Comment (nschonni) [reply package.json:13]: It wasn't a pattern I had come across before, and didn't seem to be cross-platform compatible as it relies on bash shell expansion

File: package.json [hunk 3]
```diff
@@ -5,18 +5,25 @@
...
-    "test:lint:js": "eslint --ext=.js,.jsx . --cache --report-unused-disable-directives",
-    "test:lint:sass": "stylelint \"**/*.{css,scss}\" && prettier --check \"**/*.{css,scss}\"",
+    "test": "yarn test:lint && yarn test:jest",
     "test:jest": "cross-env NODE_ENV=test jest",
-    "format": "prettier --write .",
-    "format-check": "prettier --check ."
+    "test:lint": "yarn test:lint:js && yarn test:lint:json && yarn test:lint:sass && yarn test:lint:yml",
+    "test:lint:fix": "yarn test:lint:js:fix && yarn test:lint:json:fix && yarn test:lint:sass:fix && yarn test:lint:yml:fix",
+    "test:lint:js": "eslint . --ext .js,.jsx --cache --report-unused-disable-directives",
+    "test:lint:js:fix": "yarn test:lint:js --fix",
```
Comment (takayamaki) [inline package.json:18]: [IMO]
Creating a command that can easily perform autocorrection is a nice.
However, when executed with a wildcard like `test:*`, it executes both.

I suggest naming like `format:js`.

**Comment by nschonni** [comment]
I flattened out the target names to just `lint:*` and `fix:*`

**Review by takayamaki — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (ykzts) [reply package.json:13]: I don't think `&&` is cross platform in the first place.

Comment (nschonni) [reply package.json:13]: `&&` is cross platform. Here is a litte example, first run I cut short, but run on this branch, then switched back to `main` show the error
```
PS Z:\mastodon> yarn --version
1.22.19
PS Z:\mastodon> yarn lint
yarn run v1.22.19
$ yarn lint:js && yarn lint:json && yarn lint:sass && yarn lint:yml
$ eslint . --ext .js,.jsx --cache --report-unused-disable-directives

Z:\mastodon\app\javascript\mastodon\components\status.jsx
  517:13  warning  Avoid non-native interactive elements. If using native HTML is not possible, add an appropriate role and support for tabbing, mouse, keyboard, and touch inputs to an interactive content element  jsx-a11y/no-static-element-interactions

✖ 1 problem (0 errors, 1 warning)

$ prettier --check "**/*.json"
Checking formatting...
All matched files use Prettier code style!
$ stylelint "**/*.{css,scss}" && prettier --check "**/*.{css,scss}"
PS Z:\mastodon> yarn test:lint
yarn run v1.22.19
$ ${npm_execpath} run test:lint:js && ${npm_execpath} run test:lint:sass
'${npm_execpath}' is not recognized as an internal or external command,
operable program or batch file.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
PS Z:\mastodon>
```

**Comment by nschonni** [comment]
@ykzts any more concerns with this one?

File: package.json [hunk 4]
```diff
@@ -5,20 +5,27 @@
...
-    "test:lint": "${npm_execpath} run test:lint:js && ${npm_execpath} run test:lint:sass",
-    "test:lint:js": "eslint --ext=.js,.jsx,.ts,.tsx . --cache --report-unused-disable-directives",
-    "test:lint:sass": "stylelint \"**/*.{css,scss}\" && prettier --check \"**/*.{css,scss}\"",
-    "test:typecheck": "tsc --noEmit",
-    "test:jest": "cross-env NODE_ENV=test jest",
-    "format": "prettier --write .",
-    "format-check": "prettier --check .",
-    "prepare": "husky install"
+    "test": "yarn test:lint && yarn run test:typecheck && yarn test:jest",
+    "test:typecheck": "tsc --noEmit"
```
Comment (nschonni) [inline package.json:28]: Not sure if the `test:` prefix should also be dropped here

**Comment by nschonni** [comment]
I updated the js-linting CI to call the `test:typecheck` and the expanded file globs, but I could split that out to a separate PR if it's easier to land that way

File: package.json (hunk 4 above)
Comment (renchap) [reply package.json:28]: IMO it should be dropped, or changed to `lint:typescript`, as this is more a lint than a test.

**Pull request merged**